TUM Campus App/WaitForTokenViewController.swift

```diff
@@ -42,30 +36,33 @@ extension WaitForTokenViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         button.backgroundColor = Constants.tumBlue
-        loginManager = TumOnlineLoginRequestManager(delegate: self)
-        lrzID = delegate?.getLRZ()
-        loginManager?.lrzID = lrzID
-        loginManager?.fetch()
-        checkRequest()
+        if PersistentUser.hasRequestedToken {
+            checkRequest()
+        } else {
+            button.startLoadingAnimation()
+            loginManager.fetch()
+        }
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        guard !PersistentUser.isLoggedIn else { return }
+        PersistentUser.reset()
```

Comment:
So when I login, get to the WaitForToken Screen and press the Home button (without authorising the token yet), the PersistentUser will get reset? Shouldn't the state still be .awaitingConfirmation ?

Reply:
Ah, I forgot. Does `viewWillDisapper` get called when pressing the home button?

Reply:
I checked. I doesn't get called

Reply:
Alright! 😊